app/recurring_expenses/views.py

```diff
@@ -123,53 +124,89 @@ def create_recurring_expense(request):
     return render(request, "recurring_expenses/new.html", {"form": form})
 
 
-@login_required
-def export_recurring_expenses(request):
-    export_format = request.GET.get("format", "csv")
-    if "csv" in export_format.lower():
-        filename = "recurring_expenses.csv"
-        response = HttpResponse(
-            content_type="text/csv",
-            headers={"Content-Disposition": 'attachment; filename="' + filename + '"'},
-        )
-        writer = csv.writer(response)
+_EXPORT_FORMATS = [
+    "csv",
+    "json",
+]
+
+
+def export_to_csv(expenses):
+    filename = "recurring_expenses.csv"
+    fx_rates = get_fxRate_nzd()
+    response = HttpResponse(
+        content_type="text/csv",
+        headers={"Content-Disposition": 'attachment; filename="' + filename + '"'},
+    )
+    writer = csv.writer(response)
+    writer.writerow(
+        [
+            "Active",
+            "Particulars",
+            "Amount",
+            "Currency",
+            "Amount NZD",
+            "Period (Months)",
+        ]
+    )
+    for expense in expenses:
+        amount_nzd = (1 / fx_rates[expense.currency]) * float(expense.amount)
+        expense.amount_nzd = amount_nzd
         writer.writerow(
             [
-                "Active",
-                "Particulars",
-                "Amount",
-                "Currency",
-                "Amount NZD",
-                "Period (Months)",
+                expense.active,
+                expense.particulars,
+                format(expense.amount, ".2f"),
+                expense.currency,
+                format(expense.amount_nzd, ".2f"),
+                expense.period,
             ]
         )
-        #
-        # Get expenses and amount in NZD
-        # Copied from class IndexView, maybe this should be a function?
-        # Exporting each expense to csv within this though so we don't
-        # have to run the for loop twice
-        #
-        fx_rates = get_fxRate_nzd()
-        expenses = RecurringExpense.objects.all()
-        for expense in expenses:
-            amount_nzd = (1 / fx_rates[expense.currency]) * float(expense.amount)
-            expense.amount_nzd = amount_nzd
-            writer.writerow(
-                [
-                    expense.active,
-                    expense.particulars,
-                    format(expense.amount, ".2f"),
-                    expense.currency,
-                    format(expense.amount_nzd, ".2f"),
-                    expense.period,
-                ]
-            )
-    else:
-        response = HttpResponse(status=204)
+    return response
+
 
+def export_to_json(expenses):
+    data = []
+    fx_rates = get_fxRate_nzd()
+    for expense in expenses:
+        amount_nzd = (1 / fx_rates[expense.currency]) * float(expense.amount)
+        expense.amount_nzd = amount_nzd
```

Comment:
I think this logic is duplicated for the CSV export and also for the django web view.

To keep the code DRY we should extract this logic to its own method that we run before passing the data to either export_recurring_expenses

We can do this as a future improvement though